File: Java-Core/src/main/java/presto/statement/SwitchCase.java
package presto.statement;

import presto.error.PrestoError;
import presto.error.SyntaxError;
import presto.expression.IExpression;
import presto.parser.ISection;
import presto.parser.Section;
import presto.runtime.Context;
import presto.type.IType;
import presto.utils.CodeWriter;
import presto.value.IValue;

public abstract class SwitchCase extends Section implements ISection {

	protected IExpression expression;
	StatementList statements;
	
	public SwitchCase(IExpression expression, StatementList statements) {
		this.expression = expression;
		this.statements = statements;
	}

	public abstract void checkSwitchType(Context context, IType type) throws SyntaxError;

	public IType checkReturnType(Context context) throws SyntaxError {
		return statements.check(context);
	}

	public abstract boolean matches(Context context, IValue value) throws PrestoError;

	public IValue interpret(Context context) throws PrestoError {
		return statements.interpret(context);
	}

	public abstract void caseToEDialect(CodeWriter writer);
	public abstract void caseToODialect(CodeWriter writer);
	public abstract void caseToPDialect(CodeWriter writer);
	public abstract void catchToEDialect(CodeWriter writer);
	public abstract void catchToODialect(CodeWriter writer);
	public abstract void catchToPDialect(CodeWriter writer);


}

File: Java-CodeServer/src/main/resources/js/prompto/statement/SwitchCase.js
var Section = require("../parser/Section").Section;

function SwitchCase(expression, statements) {
	Section.call(this);
	this.expression = expression;
	this.statements = statements;
	return this;
}


SwitchCase.prototype  = Object.create(Section.prototype);
SwitchCase.prototype.constructor = SwitchCase;

SwitchCase.prototype.checkReturnType = function(context) {
	return this.statements.check(context);
};

SwitchCase.prototype.interpret = function(context) {
	return this.statements.interpret(context);
};

exports.SwitchCase = SwitchCase;